### DISCUSSION TITLE
Key Management Capabilities

### DISCUSSION THREAD

**Discussion by ironhacker** [openpgpjs/openpgpjs#1460] [Q&A]
Other than generating key pairs and reading keys, does this library support any other key management features?  Specifically, is it possible to sign public keys and set trust levels?

**Comment by larabr** [comment]
Hi @ironhacker, it is possible to sign public keys using `publicKey.signAllUsers([signingKey])` or  `publicKey.signPrimaryUser([signingKey])`. Note that both function return a new key object.
If you want to certify a specific non-primary user, you can call [`User.certify()`](https://github.com/openpgpjs/openpgpjs/blob/a6c849166d7349eda8a0d34957bff0d92b2725a0/src/key/user.js#L51-L83) and add/replace the user in `publicKey.users` . 

No support for setting trust levels tho.
